springfox-core/src/main/java/springfox/documentation/service/ResolvedMethodParameter.java

```diff
@@ -31,20 +33,31 @@
 
 public class ResolvedMethodParameter {
   private final int parameterIndex;
+  private final boolean returnType;
   private final List<Annotation> annotations;
   private final String defaultName;
   private final ResolvedType parameterType;
 
   public ResolvedMethodParameter(String paramName, MethodParameter methodParameter, ResolvedType parameterType) {
     this(methodParameter.getParameterIndex(),
+        false,
         paramName,
         newArrayList(methodParameter.getParameterAnnotations()),
         parameterType);
   }
+  
+  public ResolvedMethodParameter(ResolvedType parameterType) {
+    this(0,
+        true,
+        "",
+        newArrayList(new Annotation[0]),
+        parameterType);
+  }
 
-  public ResolvedMethodParameter(int parameterIndex, String defaultName, List<Annotation> annotations, ResolvedType
+  private ResolvedMethodParameter(int parameterIndex, boolean returnType, String defaultName, List<Annotation> annotations, ResolvedType
```

Comment:
My suggestion is to use one class for all parameters, with their additional information, i.e. annotations, bean validation groups, view groups. The narrow point is [where](https://github.com/springfox/springfox/pull/1671/files#diff-2c42774555937ba3c02788ad7968e3a4R79) we will bound ResolvedParameter to ModelContext, so, after we proccessed all models, we can [pull](https://github.com/springfox/springfox/pull/1671/files#diff-8ee3acf31b9a1c7967eac49857e7a079R84) adjusted model's tree to [prepare](https://github.com/springfox/springfox/pull/1671/files#diff-3403e3edf92b23d5bac7953c2e661ee2R88) ModelRef.